### ISSUE TITLE
Machine 0.3.0-rc1 - Certificate correct?

### ISSUE THREAD

**Issue by alvinr** [docker/machine#1335]
Ran into some issues with Machine 0.3.0-rc1 and docker-py 1.2.3 rc0. It appears that the Certificate in Machine may be incorrect
```
    > from docker import Client
    > cli = Client(base_url='tcp://192.168.99.100:2376')
    > cli.containers()
    
    Traceback (most recent call last):
     File "<stdin>", line 1, in <module>
     File "/Library/Python/2.7/site-packages/docker/client.py", line 440, in containers
       res = self._result(self._get(u, params=params), True)
     File "/Library/Python/2.7/site-packages/docker/client.py", line 106, in _get
       return self.get(url, **self._set_request_timeout(kwargs))
     File "/Library/Python/2.7/site-packages/requests/sessions.py", line 477, in get
       return self.request('GET', url, **kwargs)
     File "/Library/Python/2.7/site-packages/requests/sessions.py", line 465, in request
       resp = self.send(prep, **send_kwargs)
     File "/Library/Python/2.7/site-packages/requests/sessions.py", line 573, in send
       r = adapter.send(request, **kwargs)
     File "/Library/Python/2.7/site-packages/requests/adapters.py", line 415, in send
       raise ConnectionError(err, request=request)
    requests.exceptions.ConnectionError: ('Connection aborted.', BadStatusLine('\x15\x03\x01\x00\x02\x02\n',))
```
To ensure that the environment was being accepted correct I then tried
```
    > cli = Client(**docker.utils.kwargs_from_env())
    > cli.info()

    /Library/Python/2.7/site-packages/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from     configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see     https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
     InsecurePlatformWarning
    Traceback (most recent call last):
     File "<stdin>", line 1, in <module>
     File "/Library/Python/2.7/site-packages/docker/client.py", line 729, in info
       return self._result(self._get(self._url("/info")),
     File "/Library/Python/2.7/site-packages/docker/client.py", line 106, in _get
       return self.get(url, **self._set_request_timeout(kwargs))
     File "/Library/Python/2.7/site-packages/requests/sessions.py", line 477, in get
       return self.request('GET', url, **kwargs)
     File "/Library/Python/2.7/site-packages/requests/sessions.py", line 465, in request
       resp = self.send(prep, **send_kwargs)
     File "/Library/Python/2.7/site-packages/requests/sessions.py", line 573, in send
       r = adapter.send(request, **kwargs)
     File "/Library/Python/2.7/site-packages/requests/adapters.py", line 431, in send
       raise SSLError(e, request=request)
    requests.exceptions.SSLError: no appropriate commonName or subjectAltName fields were found
```

So finally explicitly turned off tls verify which succeeded
```
    > kw = docker.utils.kwargs_from_env()
    > kw['tls'].verify = False
    > cli = Client(**kw)
    > cli.info()

    /Library/Python/2.7/site-packages/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
     InsecurePlatformWarning
    /Library/Python/2.7/site-packages/requests/packages/urllib3/connectionpool.py:768: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
     InsecureRequestWarning)
    {u'Labels': [u'provider=virtualbox'], u'NGoroutines': 16, u'LoggingDriver': u'json-file', u'HttpProxy': u'', u'DriverStatus': [[u'Root Dir', u'/mnt/sda1/var/lib/docker/aufs'], [u'Backing Filesystem', u'extfs'], [u'Dirs', u'57'], [u'Dirperm1 Supported', u'true']], u'OperatingSystem': u'Boot2Docker 1.7.0-rc1 (TCL 6.2); master : c347ab0 - Thu May 28 05:31:26 UTC 2015', u'Containers': 10, u'HttpsProxy': u'', u'MemTotal': 1046560768, u'Driver': u'aufs', u'IndexServerAddress': u'https://index.docker.io/v1/', u'InitPath': u'/usr/local/bin/docker', u'ExecutionDriver': u'native-0.2', u'OomKillDisable': True, u'SystemTime': u'2015-06-08T18:03:00.026855029Z', u'Name': u'dev2', u'RegistryConfig': {u'InsecureRegistryCIDRs': [u'127.0.0.0/8'], u'IndexConfigs': {u'docker.io': {u'Official': True, u'Name': u'docker.io', u'Secure': True, u'Mirrors': None}}}, u'NCPU': 1, u'CpuCfsQuota': True, u'Debug': True, u'ID': u'FIC4:5TB4:3VXX:G7NX:RNZK:W37O:UCTU:KFVU:QANK:IDTD:6JMP:6FKH', u'IPv4Forwarding': True, u'KernelVersion': u'4.0.4-boot2docker', u'NFd': 11, u'NoProxy': u'', u'InitSha1': u'', u'CpuCfsPeriod': True, u'MemoryLimit': True, u'SwapLimit': True, u'Images': 37, u'DockerRootDir': u'/mnt/sda1/var/lib/docker', u'NEventsListener': 0, u'ExperimentalBuild': False}
```

**Comment by shin-** [comment]
To clarify, the part relevant to machine is

`requests.exceptions.SSLError: no appropriate commonName or subjectAltName fields were found`

**Comment by nathanleclaire** [comment]
Hey @aanand Is this related to some of the recent Compose issues with SSL that you've been encountering?

**Comment by nathanleclaire** [comment]
Hm, seems to be : https://github.com/docker/compose/issues/890

**Comment by nathanleclaire** [comment]
See also https://github.com/docker/compose/issues/1484 @shin-

**Comment by aanand** [comment]
I don't think this is the same as docker/compose#890, but I did get this error ("no appropriate commonName or subjectAltName fields") when debugging that one with docker-py. It doesn't seem to happen with a VM provisioned with boot2docker-cli.

In my experience, setting `kwargs['tls'].assert_hostname = False` (and leaving `verify` set to `True`) fixed it, so the certificate isn't invalid, but it's weird that the behaviour is different from boot2docker-cli.

I did a bit of poking here: https://github.com/docker/compose/issues/890#issuecomment-106363305
